coordinator/src/node/storage.rs

```diff
@@ -112,4 +114,36 @@ impl node::Storage for NodeStorage {
         let mut conn = self.pool.get()?;
         crate::db::spendable_outputs::get_all(&mut conn)
     }
+
+    fn upsert_channel(&self, channel: Channel) -> Result<()> {
+        let mut conn = self.pool.get()?;
+        crate::db::channels::upsert(channel.into(), &mut conn)
+    }
+
+    fn get_channel(&self, user_channel_id: &str) -> Result<Option<Channel>> {
+        let mut conn = self.pool.get()?;
+        crate::db::channels::get(user_channel_id, &mut conn)
+            .map(|c| Some(c.into()))
+            .map_err(|e| anyhow!("{e:#}"))
+    }
+
+    fn all_channels(&self) -> Result<Vec<Channel>> {
```

Comment:
Maybe I'm missing something, but where is this being used?

Reply:
it's not used at the moment (besides in the tests). are you recommending to remove that function?

Reply:
I'd remove it if it is not needed as it's dead code which needs to be maintained. I guess `clippy` is not complaining because it is a public trait?

Alternatively you could annotate it with `#[cfg(test)` :)